src/hera/workflows/script.py

```diff
@@ -232,16 +236,35 @@ def _build_script(self) -> _ModelScriptTemplate:
 def _get_parameters_from_callable(source: Callable) -> Optional[List[Parameter]]:
     # If there are any kwargs arguments associated with the function signature,
     # we store these as we can set them as default values for argo arguments
-    source_signature: Dict[str, Optional[object]] = {}
+    #
+    # Take all the annotation information and make sure it is stored.
+    parameters = []
+
     for p in inspect.signature(source).parameters.values():
         if p.default != inspect.Parameter.empty and p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD:
-            source_signature[p.name] = p.default
+            default = p.default
         else:
-            source_signature[p.name] = MISSING
+            default = MISSING
+
+        param = Parameter(name=p.name, default=default)
 
-    if len(source_signature) == 0:
+        if global_config.experimental_features["script_annotations"]:
```

Comment:
We should try and reduce the complexity in terms of nesting here so that the logic becomes easier to follow.

We should be able to check for the negation of these conditions and use `continue`.

Reply:
Good call, I did that now